core/embed/rust/src/ui/layout_eckhart/firmware/vertical_menu_screen.rs

```diff
@@ -54,33 +57,24 @@ impl VerticalMenuScreen {
         self
     }
 
-    // Shift position of touch events in the menu area by an offset of the current
-    // sliding window position
-    fn shift_touch_event(&self, event: Event) -> Option<Event> {
-        match event {
-            Event::Touch(touch_event) => {
-                let shifted_event = match touch_event {
-                    TouchEvent::TouchStart(point) if self.menu.area().contains(point) => Some(
-                        TouchEvent::TouchStart(point.ofs(Offset::y(self.menu.get_offset()))),
-                    ),
-                    TouchEvent::TouchMove(point) if self.menu.area().contains(point) => Some(
-                        TouchEvent::TouchMove(point.ofs(Offset::y(self.menu.get_offset()))),
-                    ),
-                    TouchEvent::TouchEnd(point) if self.menu.area().contains(point) => Some(
-                        TouchEvent::TouchEnd(point.ofs(Offset::y(self.menu.get_offset()))),
-                    ),
-                    _ => None, // Ignore touch events outside the bounds
-                };
-                shifted_event.map(Event::Touch)
-            }
-            _ => None, // Ignore other events
+    /// Update swipe detection and buttons state based on menu size
+    pub fn update_screen(&mut self, ctx: &mut EventCtx) {
+        if !self.menu.fits_area() {
+            // Enable swipe
+            self.swipe_enabled = true;
+            self.swipe_config = SwipeConfig::new()
+                .with_swipe(Direction::Up, SwipeSettings::default())
+                .with_swipe(Direction::Down, SwipeSettings::default());
+            ctx.enable_swipe();
+
+            // Update the menu buttons state
+            self.menu.update_menu(ctx);
```

Comment:
Any reason why `update_menu` is done here but not on the ``else`` branch? Wouldn't it be nicer to do it below the `if`/`else`?